services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/S3CrtPojoConversion.java

```diff
@@ -74,16 +77,29 @@ public static CredentialsProvider createCrtCredentialsProvider(AwsCredentialsPro
                       .build();
     }
 
-    // TODO: codegen and add tests
     public static com.amazonaws.s3.model.GetObjectRequest toCrtGetObjectRequest(GetObjectRequest request) {
         com.amazonaws.s3.model.GetObjectRequest.Builder getObjectBuilder =
             com.amazonaws.s3.model.GetObjectRequest.builder()
-                                                   .key(request.key())
                                                    .bucket(request.bucket())
-                                                   .expectedBucketOwner(request.expectedBucketOwner())
                                                    .ifMatch(request.ifMatch())
                                                    .ifModifiedSince(request.ifModifiedSince())
-                                                   .ifNoneMatch(request.ifNoneMatch());
+                                                   .ifNoneMatch(request.ifNoneMatch())
+                                                   .ifUnmodifiedSince(request.ifUnmodifiedSince())
```

Comment:
Query : GetObjectRequest is updated in S3 by Codegen , what will happen if S3 add new members to GetObjectRequest  in their model?

Reply:
The tests will fail if s3 adds a new field that crt doesn't have. The tests populate all fields in the request classes and compares them with the CRT request classes, so tests fail if there are unknown fields